src/number_theory/divisors.lean

```diff
@@ -400,4 +400,16 @@ begin
   exact nat.lt_succ_of_le (nat.divisor_le (nat.mem_divisors.2 ⟨ha, h⟩))
 end
 
+/-- The factors of `n` are the prime divisors -/
+lemma prime_divisors_eq_to_filter_divisors_prime (n : ℕ) :
+n.factors.to_finset = (divisors n).filter prime :=
+begin
+    by_cases hn : n = 0,
+    { rw [hn, factors_zero, divisors_zero], simp },
+    { ext q,
+      rw [list.mem_to_finset, mem_filter, mem_divisors, mem_factors (pos_iff_ne_zero.mpr hn)],
+      rw [and_comm, and.congr_left_iff],
+      intros hq, simp only [iff_self_and, ne.def], exact imp_intro hn }
```

Comment:
```suggestion
    rcases n.eq_zero_or_pos with rfl | hn,
    { simp },
    { ext q,
      simpa [hn, hn.ne', mem_factors] using and_comm (prime q) (q ∣ n) }
```

Reply:
Thanks for this, these are a couple of beyond-absolute-beginner techniques that I hadn't picked up yet.